cmd/kubeadm/app/phases/kubeconfig/kubeconfig_test.go

```diff
@@ -180,6 +181,11 @@ func TestGetKubeConfigSpecs(t *testing.T) {
 					t.Errorf("getKubeConfigSpecs for %s Organizations is %v, expected %v", assertion.kubeConfigFile, spec.ClientCertAuth.Organizations, assertion.organizations)
 				}
 
+				// Assert EncryptionAlgorithm
+				if spec.EncryptionAlgorithm != cfg.EncryptionAlgorithm {
```

Comment:
Is the feature-gate `PublicKeysECDSA` still available in v1.31? If yes, adding a new test case to check the result when the feature gate is enabled is better. And the expected EncryptionAlgorithm should be obtained via `cfg.ClusterConfiguration.EncryptionAlgorithmType()` until the feature-gate is removed.

Reply:
@carlory i have added a second commit here that adds the missing UT